pkg/apis/core/v1beta1/types_namespacedcloudprofile.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NamespacedCloudProfile represents certain properties about a provider environment.
+type NamespacedCloudProfile struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Spec defines the provider environment properties.
+	Spec NamespacedCloudProfileSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	// Most recently observed status of the NamespacedCloudProfile.
+	Status NamespacedCloudProfileStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NamespacedCloudProfileList is a collection of CloudProfiles.
+type NamespacedCloudProfileList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Items is the list of CloudProfiles.
+	Items []NamespacedCloudProfile `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// NamespacedCloudProfileSpec is the specification of a NamespacedCloudProfile.
+// It must contain exactly one of its defined keys.
+type NamespacedCloudProfileSpec struct {
+	// CABundle is a certificate bundle which will be installed onto every host machine of shoot cluster targeting this profile.
+	// +optional
+	CABundle *string `json:"caBundle,omitempty" protobuf:"bytes,1,opt,name=caBundle"`
+	// Kubernetes contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+	// +optional
+	Kubernetes *KubernetesSettings `json:"kubernetes" protobuf:"bytes,2,opt,name=kubernetes"`
+	// MachineImages contains constraints regarding allowed values for machine images in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	MachineImages []MachineImage `json:"machineImages" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,3,opt,name=machineImages"`
```

Comment:
```suggestion
	MachineImages []MachineImage `json:"machineImages,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,3,opt,name=machineImages"`
```

Please adjust other optional fields in this file, as well.